tracer/src/Datadog.Trace/RemoteConfigurationManagement/Product.cs

```diff
@@ -24,20 +28,54 @@ protected Product()
 
         public void AssignConfigs(List<RemoteConfiguration> changedConfigs)
         {
-            List<byte[]> configurations = null;
+            List<NamedRawFile> filteredConfigs = null;
 
             foreach (var config in changedConfigs)
             {
+                var remoteConfigurationCache = new RemoteConfigurationCache(config.Path, config.Length, config.Hashes, config.Version);
+                AppliedConfigurations[remoteConfigurationCache.Path.Path] = remoteConfigurationCache;
```

Comment:
This creates a new `RemoteConfigurationCache` every time, and overwrites the existing, is that definitely correct? Also, this isn't thread safe, IIRC that's ok, but just double checking

Reply:
It's a port of this code, I believe the behaviour is the same as below:

```
            void CacheAppliedConfigurations(Product product, List<RemoteConfiguration> configurations)
            {
                foreach (var config in configurations)
                {
                    var remoteConfigurationCache = new RemoteConfigurationCache(config.Path, config.Length, config.Hashes, config.Version);

                    if (product.AppliedConfigurations.ContainsKey(config.Path.Path))
                    {
                        product.AppliedConfigurations[config.Path.Path] = remoteConfigurationCache;
                    }
                    else
                    {
                        product.AppliedConfigurations.Add(config.Path.Path, remoteConfigurationCache);
                    }
                }
            }
```

Which was being applied after the event handler. I need it before the event handle so I can apply the changes to the cache once the event handler is finished.